rct229/rules/section5/section5rule26.py

```diff
@@ -0,0 +1,274 @@
+                )["solar_heat_gain_coefficient"]
+                if bldg_scc_wwr_ratio[SCC.EXTERIOR_MIXED] > 0
+                else 0.0
+            )
+            target_shgc_res = (
+                table_G34_lookup(
+                    climate_zone,
+                    SCC.EXTERIOR_RESIDENTIAL,
+                    "VERTICAL GLAZING",
+                    wwr=bldg_scc_wwr_ratio[SCC.EXTERIOR_RESIDENTIAL],
+                )["solar_heat_gain_coefficient"]
+                if bldg_scc_wwr_ratio[SCC.EXTERIOR_RESIDENTIAL] > 0
+                else 0.0
+            )
+            target_shgc_nores = (
+                table_G34_lookup(
+                    climate_zone,
+                    SCC.EXTERIOR_NON_RESIDENTIAL,
+                    "VERTICAL GLAZING",
+                    wwr=bldg_scc_wwr_ratio[SCC.EXTERIOR_NON_RESIDENTIAL],
+                )["solar_heat_gain_coefficient"]
+                if bldg_scc_wwr_ratio[SCC.EXTERIOR_NON_RESIDENTIAL] > 0
+                else 0.0
+            )
+            target_shgc_semiheated = (
+                table_G34_lookup(
+                    climate_zone,
+                    SCC.SEMI_EXTERIOR,
+                    "VERTICAL GLAZING",
+                    wwr=bldg_scc_wwr_ratio[SCC.SEMI_EXTERIOR],
+                )["solar_heat_gain_coefficient"]
+                if bldg_scc_wwr_ratio[SCC.SEMI_EXTERIOR] > 0
+                else 0.0
+            )
+            return {
+                **data,
+                # TODO this function will likely need to be revised to RMD level later.
+                "scc_dict_b": get_surface_conditioning_category_dict(
+                    climate_zone, building_b
+                ),
+                "manual_check_required_flag": manual_check_required_flag,
+                "target_shgc_mix": target_shgc_mix,
+                "target_shgc_res": target_shgc_res,
+                "target_shgc_nores": target_shgc_nores,
+                "target_shgc_semiheated": target_shgc_semiheated,
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_b = context_item.baseline
+            scc_dict_b = data["scc_dict_b"]
+            return (get_opaque_surface_type(surface_b) == OST.ABOVE_GRADE_WALL) and (
+                scc_dict_b[surface_b["id"]] != SCC.UNREGULATED
+            )
+
+        class AboveGradeWallRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule26.BuildingRule.AboveGradeWallRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section5Rule26.BuildingRule.AboveGradeWallRule.SubsurfaceRule(),
+                    index_rmr="baseline",
+                    list_path="subsurfaces[*]",
+                    required_fields={
+                        "$.subsurfaces[*]": [
+                            "classification",
+                            "glazed_area",
```

Comment:
As I recall, in one or more of the ruleset functions, we assumed a missing `glazed_area` or `opaque_area` was zero.  We should think about whether that is what we really want and be consistent.